pkg/controller/instance/instance_controller.go

```diff
@@ -194,14 +193,11 @@ func (r *Reconciler) Reconcile(request ctrl.Request) (ctrl.Result, error) {
 }
 
 func preparePlanExecution(instance *kudov1alpha1.Instance, ov *kudov1alpha1.OperatorVersion, activePlanStatus *kudov1alpha1.PlanStatus) (*workflow.ActivePlan, *engine.Metadata, error) {
```

Comment:
i know this has been here, and doesn't affect my approval, but a 3 arity return gives me heartburn...and I don't see a way around it.

Reply:
I actually think, that this should be split into two methods: `preparePlan` and `prepareMetadata`. I don't see a real reason to have one overarching method for it. But I wanted to get @alenkacz opinion first ;)

Reply:
sure, let's split it :)